tpm/info.go

```diff
@@ -0,0 +1,159 @@
+package tpm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/go-attestation/attest"
+
+	"go.step.sm/crypto/tpm/manufacturer"
+)
+
+// Info models information about a TPM. It contains the
+// TPM version, interface, manufacturer, vendor info and
+// firmware version.
+type Info struct {
+	Version         Version         `json:"version"`
+	Interface       Interface       `json:"interface"`
+	Manufacturer    Manufacturer    `json:"manufacturer"`
+	VendorInfo      string          `json:"vendorInfo,omitempty"`
+	FirmwareVersion FirmwareVersion `json:"firmwareVersion,omitempty"`
+}
+
+// Version models the TPM specification version supported
+// by the TPM.
+type Version attest.TPMVersion
+
+func (v Version) String() string {
+	switch v {
+	case Version(attest.TPMVersion12):
+		return "TPM 1.2"
+	case Version(attest.TPMVersion20):
+		return "TPM 2.0"
+	default:
+		return "unknown"
+	}
+}
+
+// MarshalJSON marshals the version into JSON.
+func (v Version) MarshalJSON() ([]byte, error) {
+	var s string
+	switch v {
+	case Version(attest.TPMVersion12):
+		s = "1.2"
+	case Version(attest.TPMVersion20):
+		s = "2.0"
+	default:
+		s = "unknown"
```

Comment:
`s = fmt.Sprintf("unknown (%d)", v)` or in hex. whatever makes more sense to you.